lib/Serialization/SerializedModuleLoader.cpp

```diff
@@ -82,12 +82,6 @@ void forEachTargetModuleBasename(const ASTContext &Ctx,
   }
 }
 
-enum class SearchPathKind {
-  Import,
-  Framework,
-  RuntimeLibrary,
-};
-
 /// Apply \p body for each module search path in \p Ctx until \p body returns
 /// non-None value. Returns the return value from \p body, or \c None.
 Optional<bool> forEachModuleSearchPath(
```

Comment:
The only other use of `forEachModuleSearchPath` is in `collectVisibleTopLevelModuleNamesImpl` and at a cursory glance it seems like we could use `forEachModuleSearchPathContainingFile` there as well.

Reply:
I don’t think we can because we don’t know what we’re looking for. 
https://github.com/apple/swift/blob/2ef512722ae8bbcc7dff272a9db99e1dd8a33752/lib/Serialization/SerializedModuleLoader.cpp#L189-L206
just enumerates over all files in the search path.

We could probably refactor that method to recover the first level directory structure from `RelevantSearchPathForFilename` but I don’t think it’s worth it.

Reply:
Not sure how I missed the `forEachDirectyEntryPath`, that's what I get for skimming over it quickly. Bit unfortunate but it's just doing it once, so I guess it's fine 👍